File: javascript/easy/findPivotIndex.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    let sums = new Array(nums.length), sum = 0;
    for(let i = 0; i < nums.length; i++) {
        sum += nums[i];
        sums[i] = sum;
    }
    
    for(let i = 0; i < nums.length; i++) {
        const left = ((i > 0) ? sums[i-1] : 0);
        const right = ((i < nums.length-1) ? (sum - sums[i]) : 0);
        if(left === right) {
            return i;
        }
    }
    return -1;
};

File: ruby/easy/find_pivot_index.rb
# @param {Integer[]} nums
# @return {Integer}
def pivot_index(nums)
    sums = Array.new(nums.length)
    sum = 0
    for i in 0..nums.length-1
        sum += nums[i]
        sums[i] = sum
    end
    
    for i in 0..nums.length-1
        left = (i == 0) ? 0 : (sums[i-1])
        right = (i == sums.length-1) ? 0 : (sums[sums.length-1] - sums[i])
        return i if (left == right)
    end
    return -1
end
